scripts/irods/test/test_igroupadmin.py

```diff
@@ -72,6 +71,72 @@ def tearDownClass(self):
             self.admin.__exit__()
             admin_session.assert_icommand(['iadmin', 'rmuser', self.admin.username])
 
+    def test_create_user_with_password_in_igroupadmin__issue_6887(self):
+        username = 'alice'
+        # Passwords of length 42 (the current maximum) cannot be used as input for iinit
+        # until resolution https://github.com/irods/irods/issues/6913
+        password_length = 41
+        try:
+            # Create a user and assign a password of the given length.
+            passwd = (username[0]+'pass_6887_the_quick_brown_fox_jumps_over_the_lazy_dog')[:password_length]
+            self.groupadmin.assert_icommand(['igroupadmin', 'mkuser', username, passwd])
+
+            # Test that we can log in as the user, with the assigned password.
+            with session.make_session_for_existing_user(username, passwd, lib.get_hostname(), self.admin.zone_name) as user_session:
+                user_session.assert_icommand('icd')
```

Comment:
does this ... authenticate?  I think it just removes a local file...

Reply:
The `icd` won't work properly unless the session is authenticated, right?  If I remember right, I did verify it manually by stepping through the test.

Reply:
Hmm... I had this question before, but then I saw that `icd` does in fact authenticate: https://github.com/irods/irods_client_icommands/blob/0e65ee613b8be1533b667b85f9e8320574201d51/src/icd.cpp#L72

...and then didn't think about it again. However, I looked again just now and... https://github.com/irods/irods_client_icommands/blob/0e65ee613b8be1533b667b85f9e8320574201d51/src/icd.cpp#L44-L53

So... `icd` doesn't even connect to the server when you don't pass it anything. So, we should probably fix that up. `ils` or something should suffice. Good catch.

Reply:
mmm, no?    not sure.

4.2.11...
```
$ iexit full
$ ils
Enter your current iRODS password: ^C
$ icd
$ echo $?
0
```

Reply:
a deep knowledge of the code.

Reply:
On the other hand, `make_session_for_existing_user` calls `iinit`, `imkdir <session_collection>` *and* `icd <session_collection>`, so `make_session_for_existing_user` would have failed if it couldn't authenticate. Hmm...

Reply:
Ok, will change to `ils`

Reply:
For completeness...
https://github.com/irods/irods/blob/de233d4cf5a6ba41aee885000940ffadd7cff45a/scripts/irods/test/session.py#L102-L107
One of these things should be raising an `Exception` if authentication is failing, so I think this is probably fine. The `icd` line is unrelated, so you could replace it with an `ils` or just `pass`.

Reply:
I like the idea of forcing a useful, if trivial, a bit of "useful" network activity to be done, so I'll stick with the `ils` maybe...